src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,175 @@
+        tasks.add(newTask);
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + newTask);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    public static void printTasks(ArrayList<Task> tasks){
+        System.out.println("Here are the tasks in your list:");
+        for(int i = 0; i < tasks.size(); ++i) {
+            System.out.print(i + 1 + ".");
+            System.out.println(tasks.get(i).toString());
+        }
+    }
+
+    public static int getTaskIndex(String[] taskNumbers, int totalTasks) throws DukeException{
+        // From parser, in the current implementation, taskNumbers.length is always equal to 1
+        int taskIndex;
+        try {
+            taskIndex = Integer.parseInt(taskNumbers[0]);
+        } catch (Exception e){
+            throw new TaskNumberException();
+        }
+        if(taskIndex <= 0 || taskIndex > totalTasks){
+            throw new TaskNumberException();
+        }
+
+        // Minus one from task index since the array tasks is 0-based
+        taskIndex -= 1;
+        return taskIndex;
+    }
+    public static void changeTaskStatus(ArrayList<Task> tasks, String[] taskNumbers, boolean isDone) throws DukeException{
+        // From parser, taskNumbers.length is always equal to 1
+        int taskIndex = getTaskIndex(taskNumbers, tasks.size());
+
+        if(isDone){
+            if(tasks.get(taskIndex).isDone()){
+                System.out.println("This task is already marked done:");
+                System.out.println(tasks.get(taskIndex).toString());
+            } else{
+                tasks.get(taskIndex).setDone(true);
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println(tasks.get(taskIndex).toString());
+            }
+        }else{
+            if(!tasks.get(taskIndex).isDone()){
+                System.out.println("This task is already marked as not done:");
+                System.out.println(tasks.get(taskIndex).toString());
+            } else{
+                tasks.get(taskIndex).setDone(false);
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println(tasks.get(taskIndex).toString());
+            }
+        }
+    }
+
+    public static void deleteTask(ArrayList<Task> tasks, String[] taskNumbers) throws DukeException{
+        int taskIndex = getTaskIndex(taskNumbers, tasks.size());
+
+        Task taskToRemove = tasks.get(taskIndex);
+        tasks.remove(taskIndex);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("  " + taskToRemove);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    public static void main(String[] args) {
+        ArrayList<Task> tasks;
+        final File taskDataDirectory = new File("data");
+        final File taskDataFile = new File("data/task-data.txt");
```

Comment:
Consider modifying the file path such that it is a variable instead to minimize use of magic literals.